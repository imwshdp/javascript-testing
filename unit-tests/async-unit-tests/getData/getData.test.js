const { describe, test, expect, beforeEach } = require('@jest/globals');
const axios = require('axios');

const getData = require('./getData');
const { getUsers } = require('../../__mocks__/getUsers');

describe('GetData Function Tests :>>', () => {
	beforeEach(() => {
		axios.get = jest.fn().mockResolvedValue(getUsers());
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	test('Axios getting data', async () => {
		const data = await getData();

		expect(axios.get).toBeCalledTimes(1);
		expect(data).toEqual(['1', '2', '3']);

		expect(data).toMatchSnapshot();
	});
});
