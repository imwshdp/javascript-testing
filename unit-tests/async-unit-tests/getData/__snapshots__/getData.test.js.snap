// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetData Function Tests :>> Axios getting data 1`] = `
[
  "1",
  "2",
  "3",
]
`;
